### PR TITLE
[curweather] Only shown notice about unavailable weather once per session.

### PR DESCRIPTION
**Pull request by astifter**
Fix several issues in curweather widget:

- When the widget can not load weather data, it shows a notice, this notice is shown on every page load. Also the widget is plain and blank and needs to be formated properly.
  * Fix: Store notification-shown-state in session and show only once per session. (See https://github.com/friendica/friendica-addons/compare/develop...astifter:curweather_fix_header?expand=1#diff-5cd1ef666d6ce6004e4e355947ed8d10R37)
   * Fix: Restyle error widget:
Old:
screenshot_2018-07-20 friendica social network network

- Fix whitespace in PHP file.

### CODE REVIEW

File: curweather/curweather.php [hunk 1]
```diff
@@ -25,42 +25,45 @@ function getWeather( $loc, $units='metric', $lang='en', $appid='', $cachetime=0)
...
-	}
+        $cdate = PConfig::get(local_user(), 'curweather', 'last');
+        $cached = unserialize($cached);
+        if ($cdate + $cachetime > $now->getTimestamp()) {
+            return $cached;
+        }
     }
     try {
-    	$res = new SimpleXMLElement(Network::fetchUrl($url));
+            $res = new SimpleXMLElement(Network::fetchUrl($url));
```
Comment (MrPetovan) [inline curweather/curweather.php]: This indentation looks 4 spaces too wide.

File: curweather/curweather.php [hunk 2]
```diff
@@ -98,114 +101,112 @@ function curweather_network_mod_init(&$fk_app,&$b) {
...
 }
 
-
 function curweather_addon_settings_post($a,$post) {
-	if(! local_user() || (! x($_POST,'curweather-settings-submit')))
-		return;
-	PConfig::set(local_user(),'curweather','curweather_loc',trim($_POST['curweather_loc']));
-	PConfig::set(local_user(),'curweather','curweather_enable',intval($_POST['curweather_enable']));
-	PConfig::set(local_user(),'curweather','curweather_units',trim($_POST['curweather_units']));
+        if(! local_user() || (! x($_POST,'curweather-settings-submit')))
```
Comment (MrPetovan) [inline curweather/curweather.php]: The whole indentation of this function's body is twice too wide.

File: curweather/curweather.php [hunk 3]
```diff
@@ -98,114 +101,112 @@ function curweather_network_mod_init(&$fk_app,&$b) {
...
-    		'$submit' => L10n::t('Save Settings'),
-		'$header' => L10n::t('Current Weather').' '.L10n::t('Settings'),
-		'$noappidtext' => $noappidtext,
-		'$info' => L10n::t('Enter either the name of your location or the zip code.'),
-		'$curweather_loc' => [ 'curweather_loc', L10n::t('Your Location'), $curweather_loc, L10n::t('Identifier of your location (name or zip code), e.g. <em>Berlin,DE</em> or <em>14476,DE</em>.') ],
-		'$curweather_units' => [ 'curweather_units', L10n::t('Units'), $curweather_units, L10n::t('select if the temperature should be displayed in &deg;C or &deg;F'), ['metric'=>'°C', 'imperial'=>'°F']],
-		'$enabled' => [ 'curweather_enable', L10n::t('Show weather data'), $enable, '']
-	    ]);
-	return;
+        if(! local_user())
```
Comment (MrPetovan) [inline curweather/curweather.php]: The whole indentation of this function's body is twice too wide.

File: curweather/curweather.php [hunk 4]
```diff
@@ -98,114 +101,112 @@ function curweather_network_mod_init(&$fk_app,&$b) {
...
 // for accessing the API of openweathermap
 function curweather_addon_admin_post (&$a) {
-	if(! is_site_admin())
-	    return;
-	if ($_POST['curweather-submit']) {
-	    Config::set('curweather','appid',trim($_POST['appid']));
-	    Config::set('curweather','cachetime',trim($_POST['cachetime']));
-	    info(L10n::t('Curweather settings saved.'.EOL));
-	}
+        if(! is_site_admin())
```
Comment (MrPetovan) [inline curweather/curweather.php]: The whole indentation of this function's body is twice too wide.

**Comment by astifter** [comment]
@MrPetovan But the changes with the session are okay?

**Comment by MrPetovan** [comment]
My only problem with it is that you don't have a consistent timeout for showing the error again. But maybe it isn't a problem.

File: curweather/curweather.php [hunk 5]
```diff
@@ -25,42 +25,46 @@ function getWeather( $loc, $units='metric', $lang='en', $appid='', $cachetime=0)
     $cached = Cache::get('curweather'.md5($url));
     $now = new DateTime();
     if (!is_null($cached)) {
-	$cdate = PConfig::get(local_user(), 'curweather', 'last');
-	$cached = unserialize($cached);
-	if ($cdate + $cachetime > $now->getTimestamp()) {
-	    return $cached;
-	}
+        $cdate = PConfig::get(local_user(), 'curweather', 'last');
```
Comment (tobiasd) [inline curweather/curweather.php]: Can you please [use tabs](https://github.com/friendica/friendica/blob/develop/doc/Developers-Intro.md#coding-standards) for indentation and not spaces.

Comment (MrPetovan) [reply curweather/curweather.php]: The whole file uses spaces, and it wasn't supposed to be a complete formatting do-over. In this case I'd rather have the same indentation file-wise even if it's spaces.

Comment (astifter) [reply curweather/curweather.php]: @tobiasd I will next time. Shall I change that file to tabs or leave it alone?

Comment (MrPetovan) [reply curweather/curweather.php]: Please change indentation to tabs, the goal is to have a unified code base with the same standards everywhere.

**Pull request merged**